python/tk_multi_workfiles/file_item.py

```diff
@@ -228,7 +228,7 @@ def _get_versions(self):
         :returns:   A dictionary of {version:FileItem} containing a map of all other
                     versions of this file
         """
-        return self._versions
+        return {k: self._versions[k] for k in sorted(self._versions)}
```

Comment:
I had a quick look at the OrderedDict object since I never used it before. I am a bit confused with the documentation so I am not sure what to do with it.
Maybe using it from inside the _set_versions function.

Anyway, I get your point, you want to secure the data structure as early as possible. What about moving your code to the _set_versions function instead? This way, the dictionary instance from this object would always be sorted and this would avoid building a new one at each  _get_versions call?

Reply:
+1 on sorting on the @versions.setter. It's a good use case for a setter -- formatting the data for usage.

Reply:
I agree sorting should be done on UI's end. I don't have much context about the UI of this section and also don't know how ofter this FileItem objects is updated, but we can explore how to use [cached_property](https://docs.python.org/3/library/functools.html#functools.cached_property) on UI's end to perform the sorting just once.